app/main.py

```diff
@@ -1,3 +1,76 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"{self.__class__.__name__}: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}(km={self.km})"
+
+    def __add__(self, distance2: Distance | int | float) -> Distance:
+        if isinstance(distance2, (int, float)):
+            return Distance(self.km + distance2)
+        elif isinstance(distance2, Distance):
+            return Distance(self.km + distance2.km)
+        else:
+            raise TypeError
```

Comment:
This cheks are not needed. You specify that your code works only with int, float and Distance
```suggestion
        return Distance(self.km + distance2.km)
```

Reply:
Fin in all places

Reply:
But isn't it a good practice to add some error handling? I can still pass a string or a list as an argument and get an error, like:
```
TypeError: unsupported operand type(s) for +: 'float' and 'str'
```

And the cause of this problem is not really obvious